OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,411 @@
+                        combatant.ID,
+                        string.Join("", changed.Select((fi) => FormatFieldChange(fi, combatant))));
+                }
+            }
+
+            // Any combatants no longer in memory, signify that they were removed
+            foreach (var ID in combatantStateMap.Keys)
+            {
+                // If we're filtering, only consider removing those in the filters
+                if (filter.Length > 0 && !filter.Contains(ID))
+                {
+                    continue;
+                }
+                if (!combatants.Any((c) => c.ID == ID))
+                {
+                    combatantStateMap.TryRemove(ID, out var _);
+                    WriteLine(CombatantMemoryChangeType.Remove, ID, "");
+                }
+            }
+        }
+
+        private string FormatFieldChange(FieldInfo info, Combatant combatant, bool skipDefaultValues = false)
+        {
+            var value = info.GetValue(combatant);
+
+            if (value == null)
+            {
+                if (skipDefaultValues)
+                {
+                    return string.Empty;
+                }
+                return $"|{info.Name}|NULL";
+            }
+
+            if (skipDefaultValues && value.Equals(CombatantChangeCriteria.DefaultValues[info.FieldType]))
+            {
+                return string.Empty;
+            }
+
+            if (info.Name == "PCTargetID" || info.Name == "NPCTargetID" || info.Name == "BNpcNameID" || info.Name == "BNpcID" || info.Name == "TargetID" || info.Name == "OwnerID" || info.Name == "CastTargetID")
+            {
+                return $"|{info.Name}|0x{value:X}";
+            }
+
+            if (info.FieldType.IsEnum)
+            {
+                return $"|{info.Name}|{Convert.ChangeType(value, Enum.GetUnderlyingType(info.FieldType))}";
+            }
+            return $"|{info.Name}|{value}";
+        }
+
+        private void WriteLine(CombatantMemoryChangeType type, uint combatantID, string info)
+        {
+            var line = $"{type}|{combatantID:X8}{info}";
+            logWriter(line, ffxiv.GetServerTimestamp());
+        }
+
+        private unsafe void MessageReceived(string id, long epoch, byte[] message)
+        {
+            fixed (byte* buffer = message)
+            {
+                uint actorID = *(uint*)&buffer[offsetHeaderActorID];
+                uint loginID = *(uint*)&buffer[offsetHeaderLoginUserID];
+                // Only check if we're not looking at a packet that's for just us
+                if (actorID != loginID)
+                {
+                    DateTime serverTime = ffxiv.EpochToDateTime(epoch);
+                    // Also only check if we're beyond the default delay for this ID, or if this ID doesn't exist yet
+                    if (!combatantStateMap.ContainsKey(actorID) || (serverTime - combatantStateMap[actorID].lastUpdated).TotalMilliseconds > CombatantChangeCriteria.InCombatDelayDefault)
```

Comment:
Sorry if I'm misunderstanding, but is this saying that this will only check the actor if it's new or the total time is greater than `InCombatDelayDefault` (1 second)? The only other way into `CheckCombatants` is via `PollCombatants`, so how does it ever "Check for partial change lines with a delay less than the default delay" as the comment above says?  Sorry if I am just misreading this, but I am not sure if I am misunderstanding.

Reply:
The goal of this is to trigger a 261 line if there's any network packet that pertains to the combatant, but avoid spamming memory reads just because e.g. 8 packets for a raidwide happen on the same frame. So the logic is:
```
if this packet isn't for the current player's ID
  if this ID hasn't already had a line written, write one
  else if the timestamp of this packet is at least one second after the last time a line was written for this ID
    if this combatant has a valid change, write a line
```

Reply:
And then writing line behavior is "if there's a change in DefaultCheckFieldNames, write the full combatant, otherwise write only fields that have changed past the threshold".

I guess I had misread the threshold as something to write out sooner than the 1 second `InCombatDelayDefault` rather than as something to filter out changes during that polling interval.

Reply:
I suppose for situations where 1 second is too long we still need to rely on `getCombatants`? One situation I am thinking of is something like TOP P2 where a player has a lower response time between M/F `ModelStatus === 0` and mechanic resolution. (side note: I am not actually sure if those combatants exist prior to those mechanics or not, but it's just an example scenario)

Reply:
If something needs to be checked more often, then it can be added to do so. For example, if you want `ModelStatus` changes to be reflected immediately, that can be done, as it's being done for `WeaponId` and `TransformationID`.

If additional fields are going to be added like this I might slightly refactor the code around those checks to make it easier to add/remove in the future.